src/mtr.cpp

```diff
@@ -0,0 +1,211 @@
+/*
+  Adplug - Replayer for many OPL2/OPL3 audio file formats.
+  Copyright (C) 1999 - 2003 Simon Peter, <dn.tlp@gmx.net>, et al.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+  mtr.cpp - MTR loader by Dmitry Smagin <dmitry.s.smagin@gmail.com>
+*/
+
+#include <stdio.h>
+#include <cstring>
+#include "mtr.h"
+#include "debug.h"
+
+/* -------- Public Methods -------------------------------- */
+
+CPlayer *CmtrLoader::factory(Copl *newopl) {
+    return new CmtrLoader(newopl);
+}
+
+bool CmtrLoader::load(const std::string &filename, const CFileProvider &fp) {
+    binistream *f = fp.open(filename);
+    if (!f)
+        return false;
+
+    unsigned int i, j, k, t = 0;
+    ninstruments = 0;
+
+    char header[51] = {0};
+    char mtitle[21] = {0};
+    int nvoices, ndigvoices, npatterns, orderlen,
+        timervalue = 0x428F, restart, flength;
+
+    f->readString(header, 50);
+
+    if (!strncmp(header, "MTRAC ", 6)) {
```

Comment:
Documentation says "MTRACK ". Is the documentation incorrect, if so, add a comment please

Reply:
It is incorrect.

Reply:
We might get lucky to find a user with a rare file from a version that is not in the modland.com archives

Reply:
I've checked http://modland.com/pub/modules/Ad%20Lib/Master%20Tracker/ and it contains the same MTR modules that were packed with the original Master Tracker. I've tested them all with this loader, and they play identically with MT.